src/routes/goals/handlers.js

```diff
@@ -256,3 +259,38 @@ export async function mergeGoalHandler(req, res) {
     await handleErrors(req, res, err, `${logContext}:MERGE_GOAL`);
   }
 }
+
+export async function getSimilarGoalsForRecipient(req, res) {
+  const recipientId = parseInt(req.params.recipientId, DECIMAL_BASE);
```

Comment:
Do we need to check permissions here? Or maybe it's on the FE. For example we have a user visiting the page who can't perform a merge then we don't need to do any this.

Reply:
we check here, a little below this line

Reply:
Just want to make sure we are not doing processing we don't need to.

Reply:
A read permission check happens a few lines down:

https://github.com/HHS/Head-Start-TTADP/blob/76beb59e008198cb88105da49411718c9ad09654/src/routes/goals/handlers.js#L278-L289

Since this handler is not responsible for any mutation of a Goal, I don't think we need to check for write permissions here - probably should happen in an actual merge handler.